src/transformStateWithClones.js

```diff
@@ -8,6 +8,40 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateCopy = { ...state };
+  const arrOfActions = [];
+
+  for (const action of actions) {
+    stateCopy = { ...stateCopy };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+
+        break;
+
+      case 'clear':
+        for (const key of Object.getOwnPropertyNames(stateCopy)) {
+          delete stateCopy[key];
+        }
+
+        break;
+
+      default:
```

Comment:
throw some error here

Reply:
Ok, but for what?

Reply:
so the user can know - he entered the wrong action type.
in your case, you just return a string, but the user may just call this function even without looking at what it returns 😉